kernel/kernel-api/src/test/scala/io/delta/kernel/utils/CloseableIteratorSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.utils
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers.contain
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+
+import java.util
+
+class CloseableIteratorSuite extends AnyFunSuite {
+
+  test("test take while with all elements matching filter") {
+    val res = new util.ArrayList[Int]();
+    new IntArrayIterator(List(1, 2, 3, 2, 2))
+      .takeWhile((input: Int) => input > 0)
+      .forEachRemaining((input: Int) => res.add(input))
+    res should contain theSameElementsAs List(1, 2, 3, 2, 2)
+  }
+
+  test("test take while with no elements matching filter") {
+    val res = new util.ArrayList[Int]();
+    new IntArrayIterator(List(1, 2, 3, 2, 2))
+      .takeWhile((input: Int) => input < 0)
+      .forEachRemaining((input: Int) => res.add(input))
+    assert(res.isEmpty)
+  }
+
+  test("test take while with some elements matching filter") {
```

Comment:
we want to check that the last two `2`s are not even visited. not only are they excluded from the returned output, but they aren't even checked.

you can do this with a custom iterator that throws an exception, e.g. imagine i have values 1 2 3 4 5 6.

first, i have an iterator that throws if it sees a values >= 5.

then i have an iterator that takesWhile value is <= 4

the value = 5 should never be seen and the exception should never be thrown.